Source/Core/Resource.js

```diff
@@ -212,21 +212,32 @@ function combineQueryParameters(q1, q2, preserveQueryParameters) {
   return result;
 }
 
+// QueryParameters must be a one-deep object whose values are a primtive or array thereof
+/** @typedef {string | number | boolean} QueryValue */
+/** @typedef {Object.<string, QueryValue | QueryValue[] | undefined>} Resource.QueryParameters */
+
+/**
+ * @typedef {Object} Resource.ConstructorOptions
+ *
+ * Initialization options for the Resource constructor
+ *
+ * @property {String} url The url of the resource.
+ * @property {Resource.QueryParameters} [queryParameters] An object containing query parameters that will be sent when retrieving the resource.
```

Comment:
QueryParameters is not showing up properly in the doc, hopefully an easy fix.

![image](https://user-images.githubusercontent.com/1495196/83272055-b27f2780-a198-11ea-8c22-2b429abebdfa.png)

Reply:
This is a great point.  Since this is a type alias, basically for Object<string, string|num|bool|(string|num|bool)[]|undefined>, there's really nothing to print in the template.  I should have put some text description in there, at least.

Reply:
I went ahead and added a one-liner so it's not blank, at least.  If you want to flesh it out feel free.

Reply:
I think it's going to be really confusing for non TS users reading the doc to understand what they are supposed to pass here.  What is the fundamental issue you are trying to prevent here, people passing an object that has more than just primitive values?  Technically users can pass anything that has a `toString` function, right?  So wouldn't `Object` still be the best option here no matter what?